server/resp/src/test/java/org/infinispan/server/resp/RespBRPOPTest.java

```diff
@@ -0,0 +1,534 @@
+package org.infinispan.server.resp;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.spy;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.infinispan.commons.util.ByRef;
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.notifications.cachelistener.CacheNotifier;
+import org.infinispan.notifications.cachelistener.CacheNotifierImpl;
+import org.infinispan.notifications.cachelistener.ListenerHolder;
+import org.infinispan.server.resp.commands.list.blocking.BPOP;
+import org.infinispan.test.TestingUtil;
+import org.mockito.stubbing.Answer;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+
+import io.lettuce.core.KeyValue;
+import io.lettuce.core.RedisCommandExecutionException;
+import io.lettuce.core.RedisFuture;
+import io.lettuce.core.api.async.RedisAsyncCommands;
+import io.lettuce.core.api.sync.RedisCommands;
+
+/**
+ * Class for blocking commands tests.
+ *
+ * @author Vittorio Rigamonti
+ * @since 15.0
+ */
+@Test(groups = "functional", testName = "server.resp.RespBRPOPTest")
+public class RespBRPOPTest extends SingleNodeRespBaseTest {
```

Comment:
We could also abstract something using a strategy. Having the default blocking test suite, where each concrete type implements the listener install and how to send the operation.

Reply:
Yeah, I definitely think we should abstract something for the tests here as this is a ton of almost identical code.

Reply:
With the creation of `RespBxPOPTest` are the `RespBRPOPTest` and `RespBLPOPTest` needed? It seems the `RespBxPOPTest` covers everything from the other two.

Reply:
sorry I missed this comment. Yeah you're right, I had leaved behind some tests because they differs in the two versions, but they can be unified adding some conditionals.
done that and removed the old files